src/main/java/com/google/codeu/servlets/UserFormServlet.java

```diff
@@ -60,25 +63,43 @@ protected Datastore createDatastore() {
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
 
-    Map<String, Long> userTeaData = new HashMap<String, Long>();
+        Map<String, Long> userTeaData = new HashMap<String, Long>();
 
-    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd");
-    LocalDate localDate = LocalDate.now();
-    String date = dtf.format(localDate);
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+        LocalDate localDate = LocalDate.now();
+        String date = dtf.format(localDate);
 
-    UserService userService = UserServiceFactory.getUserService();
-    if (!userService.isUserLoggedIn()) {
-      response.setStatus(401);
-      response.getWriter().println("Error: Unauthorized access");
-      return;
-    }
-    String username = userService.getCurrentUser().getEmail();
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
```

Comment:
I believe that this is a temp solution to https://github.com/codeu-spring-2019-team-9/codeu-spring-2019-team-9/pull/17#discussion_r270426355
But I am still stuck on trying to figure out how to do a hard failure with a 401 status code on the client side,
I was thinking of using a try and catch block and when it catches that the user is not logged in, it would then emitt error 401. But I am currently stuck on seeing how I can do that in JavaScript if that is the correct way of doing this.

Reply:
Please see the documentation for the JavaScript `fetch` function:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

As well as for the `Response` object that is returned as the result of the `Promise`:
https://developer.mozilla.org/en-US/docs/Web/API/Response

And this documentation for the Promise API:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
https://developers.google.com/web/fundamentals/primers/promises

In particular, from the documentation for `fetch`, note:

> The `fetch()` method takes one mandatory argument, the path to the resource you want to fetch. It returns a Promise **that resolves to the `Response` to that request, _whether it is successful or not_**. 

And, from the documentation for `Response`, please note the properties `ok`, `status`, and `statusText`. Here is a snippet of my own that should help:

```
     var handleSuccessfulResponse = function(response) {
        console.log('Reponse was successfull');
        // ... do stuff with valid response ...
     };

     var handleErrorResponse = function(response) {
        console.log('Response was unsuccessful'); 
        console.log('Status code: ' + response.status);
        console.log('Status message: ' + response.statusText);
        // ... do stuff with error response ...
     };

     // A version of "fetch" that returns a Promise that splits
     // responses so that "then" only receives successful responses
     // and "catch" only receives unsuccessful responses.
     var fetchWithErrorHandling = function(url) {
       return fetch(url).then(function(response) {
         if (!response.ok) {
           console.log('Response is not OK');
           throw response;
         }
         console.log('Response is OK');
         return response;
       });
     };
       
    // ...
    var eventualResponse = fetchWithErrorHandling(someUrl);
    eventualResponse.then(handleSuccessfulResponse);
    eventualResponse.catch(handleErrorResponse);
    // ...   
```